controllers/workspace/devworkspace_controller.go

```diff
@@ -281,9 +281,15 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 	workspace.Spec.Template = *flattenedWorkspace
 
-	err = ssh.AddSshAgentPostStartEvent(&workspace.Spec.Template)
-	if err != nil {
-		return r.failWorkspace(workspace, "Failed to add ssh-agent post start event", metrics.ReasonWorkspaceEngineFailure, reqLogger, &reconcileStatus), nil
+	if workspace.Config.EnableExperimentalFeatures != nil && *workspace.Config.EnableExperimentalFeatures {
+		if needsSSHAgentPostStartEvent, err := ssh.NeedsSSHPostStartEvent(clusterAPI, workspace.Namespace); err != nil {
+			// TODO: Should we fail the workspace? Or log the error and continue on?
```

Comment:
+1
Given the fact that this PR aims to prevent breaking any workspaces that would otherwise work fine prior to the SSH agent postStart event, we should probably just log an error rather than failing the workspace.